### PR TITLE
pcapgo: add read/write support for Decryption Secrets Block (DSB)

### PR DESCRIPTION
**Pull request by stv0g**
Prebiously tracked in Google repo as: https://github.com/google/gopacket/pull/1042

Originally submitted by: @cfc4n

### CODE REVIEW

File: pcapgo/ngread_dsb_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package pcapgo
+
+import (
+	"fmt"
+	"github.com/google/gopacket"
```
Comment (mosajjal) [inline pcapgo/ngread_dsb_test.go:5]: this needs to change to "github.com/gopacket/gopacket" for the tests to run

**Comment by stv0g** [comment]
Thanks @mosajjal,

I have fixed the imports now.

**Comment by mosajjal** [comment]
Codes and tests look good. just need to convert the pcap files used in testing to the actual bytes of each packet to make it consistent with other tests. refer to previous PRs or _test files.

**Comment by stv0g** [comment]
Thanks @mosajjal for the review.

Could you elaborate a bit? It seems like also the other tests read-on the pcapng files:

https://github.com/gopacket/gopacket/blob/ee81b8c880dac66412e96a9f0ef45186380edc51/pcapgo/ngread_test.go#L69

**Comment by mosajjal** [comment]
to me the preference is to use byte stream rather than a `pcap` file to test out a particular packet. but if that's not possible for this use case that's fine too. is there any other changes needed for this or is it ready for merge now? Happy to merge it anytime :)

**Comment by stv0g** [comment]
I think we are better of with the PCAPng files here as we dont test against packet data but rather against the DSB sections within the file.

From my side its ready to merge.

**Pull request merged**